### PR TITLE
types: add types for createInstance

### PR DESCRIPTION
**Pull request by AVaksman**
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Towards #748

### CODE REVIEW

**Comment by skuruppu** [comment]
@AVaksman if you could check the build failures, that would be great.

File: src/index.ts [hunk 1]
```diff
@@ -434,15 +465,15 @@ class Spanner extends GrpcService {
         {
           name: formattedName,
           displayName: shortName,
+          nodeCount: config.nodes || 1,
```
Comment (stephenplusplus) [inline src/index.ts]: Is it possible the user would want `0`?

Comment (AVaksman) [reply src/index.ts]: [Docs](https://cloud.google.com/spanner/docs/instances#node_count)
`0` is not a valid value. The request with `node_count` as `0` will be rejected with 
`INVALID_ARGUMENT: Invalid CreateInstance request.` error.

**Comment by AVaksman** [comment]
It seems to be a bug with coverage report.
closing this PR in favor of #805 which has the same exact changes

**Pull request closed without merging**